types/underscore/underscore-tests.ts

```diff
@@ -12,13 +12,14 @@ _.map({ one: 1, two: 2, three: 3 }, (value, key) => value * 3);
 let plucked: string[] = _.map([{key: 'apples'}, {key: 'oranges'}], 'key');
 
 //var sum = _.reduce([1, 2, 3], (memo, num) => memo + num, 0);    // https://typescript.codeplex.com/workitem/1960
-var sum = _.reduce<number, number>([1, 2, 3], (memo, num) => memo + num, 0);
-sum = _.reduce<number, number>([1, 2, 3], (memo, num) => memo + num); // memo is optional #issue 5 github
-sum = _.reduce<string, number>({'a':'1', 'b':'2', 'c':'3'}, (memo, numstr) => memo + (+numstr));
```

Comment:
I removed this test because it was using this overload in a way that is not always accurate. In the case of a collection of a single item (example below), the collection type is returned rather than the type that TS is told will be returned.

```TS
// TS is told that the below will return a number, but if the collection contains only one item it returns
// that item without ever calling iteratee
// the below results in a value of '1' when run in JS
_.reduce<string, number>({'a':'1'}, (memo, numstr) => memo + (+numstr));
```

Reply:
When rewriting this test to take `memo` as a type union, TS forced me to also take steps to make sure `memo` was a number, which I thought was just a formality but then I thought about it a bit more and found that when I actually run this test as it was originally written I actually get `'123'` as a result rather than a number. Yay extra type checking I guess 😄